src/main/java/travel/travelapplication/user/application/UserService.java

```diff
@@ -1,178 +1,177 @@
 import travel.travelapplication.plan.domain.Plan;
 import travel.travelapplication.user.domain.User;
-import travel.travelapplication.userplan.domain.UserPlan;
-import travel.travelapplication.userplan.repository.UserPlanRepository;
 import travel.travelapplication.user.repository.UserRepository;
-
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
+import travel.travelapplication.userplan.repository.UserPlanRepository;
 
 @RequiredArgsConstructor
 @Service
 public class UserService {
 
-    private final UserRepository userRepository;
-    private final TagRepository tagRepository;
-    private final UserPlanRepository userPlanRepository;
-
-    public void save(User user) {
-        userRepository.save(user);
-    }
-
-    public void updateUserName(String email, String username) throws IllegalAccessException {
-        User user = userRepository.findByEmail(email)
-                .orElse(null);
-
-        if (user != null) {
-            User updatedUser = User.builder()
-                    .name(username)
-                    .email(user.getEmail())
-                    .accessToken(user.getAccessToken())
-                    .userPlans(user.getUserPlans())
-                    .savedPlans(user.getSavedPlans())
-                    .tags(user.getTags())
-                    .likedPlaces(user.getLikedPlaces())
-                    .role(user.getRole())
-                    .build();
-
-            user.update(updatedUser);
-            save(user);
-        } else {
-            throw new IllegalAccessException("존재하지 않는 사용자입니다.");
-        }
-    }
-
-    public void addTag(User user, List<String> tagIds) throws IllegalAccessException {
-        List<ObjectId> objectIdList = tagIds.stream()
-                .map(ObjectId::new)
-                .collect(Collectors.toList());
-        List<Tag> tags = tagRepository.findAllById(objectIdList);
-
-        if (user != null) {
-            User updatedUser = User.builder()
-                    .name(user.getName())
-                    .email(user.getEmail())
-                    .accessToken(user.getAccessToken())
-                    .userPlans(user.getUserPlans())
-                    .savedPlans(user.getSavedPlans())
-                    .tags(tags)
-                    .likedPlaces(user.getLikedPlaces())
-                    .role(user.getRole())
-                    .build();
-
-            user.update(updatedUser);
-            save(user);
-        } else {
-            throw new IllegalAccessException("존재하지 않는 사용자입니다.");
-        }
+  private final UserRepository userRepository;
+  private final TagRepository tagRepository;
+  private final UserPlanRepository userPlanRepository;
+
+  public void save(User user) {
+    userRepository.save(user);
+  }
+
+  public void updateUserName(String email, String username) throws IllegalAccessException {
+    User user = userRepository.findByEmail(email)
+        .orElse(null);
```

Comment:
값을 변경하고자 하는 필드만 setter를 이용해서 변경할 수 있도록 하고, 이를 도메인에서 수행하도록 하는 것은 어떤지 상의하면 좋을 것 같습니당.

```suggestion
public class User {
  String username;

  public void updateUserName(String username) {
    this.username=username;
  }
}

public class UserService {
  public void updateUserName(String email, String username) {
    User user = userRepository.findByEmail(email);
    user.updateUserName(username);
  }
}